xrpl/asyncio/transaction/main.py

```diff
@@ -202,19 +202,32 @@ def _validate_account_xaddress(
     """
     Mutates JSON-like dictionary so the X-Address in the account field is the classic
     address, and the tag is in the tag field.
+
+    Args:
+        json: JSON-like dictionary with transaction data or similar
+        account_field: the field of `json` that may contain an X-Address
+        tag_field: the field of `json` that may contain a source or destination tag
+
+    Raises:
+        XRPLException: if both an X-Address containing a tag and a tag field are
+            provided and they do not match.
     """
     if is_valid_xaddress(json[account_field]):
         account, tag, _ = xaddress_to_classic_address(json[account_field])
         json[account_field] = account
-        if json[tag_field] and json[tag_field] != tag:
+        if tag_field in json and json[tag_field] != tag:
```

Comment:
Would you mind adding a test case for this fix?

Reply:
Added